src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  function callback(prevItem: GroupsMap<T>, currentItem: T): GroupsMap<T> {
+    const currentKey: string = String(currentItem[key]);
+    const groupedObj: GroupsMap<T> = prevItem;
```

Comment:
Why can’t you use prevItem?

```suggestion
```

Reply:
Because of eslint error : 'Assignment to property of function parameter 'prevItem''))